scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ConnectionEvictedException.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Sierra Wireless and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ ******************************************************************************/
+package org.eclipse.californium.scandium.dtls;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Raised when a connection is evicted from
+ * {@link ResumptionSupportingConnectionStore}
+ * 
+ * @since 2.3
+ */
+public class ConnectionEvictedException extends DtlsException {
+
+	private static final long serialVersionUID = 1L;
+
+	public ConnectionEvictedException(String message, InetSocketAddress peer) {
```

Comment:
The peer maybe `null`. I'm not sure, if that would happen also before. At least, I would prefer to add now javadoc to DtlsException including the version, when `null` is introduced.

Reply:
I will add the javadoc. But my guess is a `null` peer address could maybe already happen. So not sure it makes sense to add the `since version`. By the way nothing in the javadoc says that peer address can not be null too.

What do you think ? Should I add the `since version` ?

(This is not part of this PR, but my guess is that the correct API could be to have `Connection`(or something like this) instead of peer address.)

Reply:
(I added the javadoc without since for now)

Reply:
> By the way nothing in the javadoc says that peer address can not be null too.

There is also no indication, that `null` maybe returned.

> What do you think ? Should I add the since version ?

That makes it clear, that at least from that version on, `null` maybe returned.
If I would get "new" NullPointerExceptions, I would appreciate, to read, that this is cased by a change with the 2.3.

Reply:
At least, if you add a `null`check (NullPointerException) and run `mvn clean test`, the tests are not failing.
In the most scenarios, I was able to check (without wasting too much time for this), the exceptions results from receiving messages, in which case the current address is available.

If `null`is already returned, existing code would be aware of that.

The comment just helps in cases, where until now, no `null`was returned to already existing code. Just to mention: I didn't found, that californium calls the getter at all.

Reply:
I will add the since version warn because I don't really want to take more time to talk about the color of the bike.

Reply:
Done